src/Renderer/OrientedImageMaterial.js

```diff
@@ -0,0 +1,113 @@
+import * as THREE from 'three';
+import textureVS from './Shader/ProjectiveTextureVS.glsl';
+import textureFS from './Shader/ProjectiveTextureFS.glsl';
+import project_pars_vertex from './Shader/Chunk/project_pars_vertex.glsl';
+import projective_texturing_vertex from './Shader/Chunk/projective_texturing_vertex.glsl';
+import projective_texturing_pars_vertex from './Shader/Chunk/projective_texturing_pars_vertex.glsl';
+import projective_texturing_pars_fragment from './Shader/Chunk/projective_texturing_pars_fragment.glsl';
+import Capabilities from '../Core/System/Capabilities';
+
+THREE.ShaderChunk.projective_texturing_vertex = projective_texturing_vertex;
+THREE.ShaderChunk.projective_texturing_pars_vertex = projective_texturing_pars_vertex;
+THREE.ShaderChunk.projective_texturing_pars_fragment = projective_texturing_pars_fragment;
+THREE.ShaderChunk.project_pars_vertex = project_pars_vertex;
+
+
+// adapted from unrollLoops in WebGLProgram
+function unrollLoops(string, defines) {
```

Comment:
This can be added in `ShaderUtils`

Reply:
Actually I added it in #867 so if it gets merged before this one you can remove it